sdks/python/apache_beam/utils/profiler.py

```diff
@@ -67,3 +67,77 @@ def __exit__(self, *args):
           self.profile, stream=s).sort_stats(Profile.SORTBY)
       self.stats.print_stats()
       logging.info('Profiler data: [%s]', s.getvalue())
+
+
+class MemoryReporter(object):
+  """A memory reporter that reports the memory usage and heap profile.
+  Usage:
+    mr = MemoryReporter(interval_sec=30.0)
+    mr.start()
+    while ...
+      <do something>
+      # this will report continuously with 30 seconds between reports.
+    mr.stop()
+
+  Or simply:
+    with MemoryReporter(interval_sec=100):
+      <do some thing>
+
+  Also it could report on demand without continuous reporting.
+    mr = MemoryReporter()  # default interval 60s but not activated.
+    <do something>
+    mr.report_once()
+  """
+
+  def __init__(self, interval_sec=60.0):
+    # guppy might not have installed. Set up the reporter only when guppy is
```

Comment:
Could you add a comment line about, `guppy` not being available on all platforms.